shell/platform/linux/public/flutter_linux/fl_application.h

```diff
@@ -0,0 +1,89 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_PUBLIC_FLUTTER_LINUX_FL_APPLICATION_H_
+#define FLUTTER_SHELL_PLATFORM_LINUX_PUBLIC_FLUTTER_LINUX_FL_APPLICATION_H_
+
+#if !defined(__FLUTTER_LINUX_INSIDE__) && !defined(FLUTTER_LINUX_COMPILATION)
+#error "Only <flutter_linux/flutter_linux.h> can be included directly."
+#endif
+
+#include "fl_plugin_registry.h"
+#include "fl_view.h"
+
+#include <gmodule.h>
+#include <gtk/gtk.h>
+
+G_BEGIN_DECLS
+
+G_MODULE_EXPORT
+G_DECLARE_DERIVABLE_TYPE(FlApplication,
+                         fl_application,
+                         FL,
+                         APPLICATION,
+                         GtkApplication);
+
+/**
+ * FlApplicationClass:
+ * @register_plugins: invoked when plugins should be registered.
+ */
+struct _FlApplicationClass {
+  GtkApplicationClass parent_class;
+
+  /**
+   * FlApplication::register_plugins:
+   * @application: the application
+   * @registry: registry to use.
+   *
+   * The ::register_plugins signal is emitted when plugins can be registered.
+   */
+  void (*register_plugins)(FlApplication* application,
+                           FlPluginRegistry* registry);
+};
+
+/**
+ * FlApplication:
+ *
+ * #Flutter-based application with the GTK embedder.
+ *
+ * Provides default behaviour for basic Flutter applications.
+ */
+
+/**
+ * fl_application_new:
+ * @application_id: (allow-none): The application ID or %NULL.
+ * @flags: The application flags.
+ *
+ * Creates a new Flutter-based application.
+ *
+ * Returns: a new #FlApplication
+ */
+FlApplication* fl_application_new(const gchar* application_id,
+                                  GApplicationFlags flags);
+
+/**
+ * fl_application_set_default_window_title:
+ * @application: an #FlApplication.
+ * @window_title: window title text.
+ *
+ * Sets the title to apply to Flutter windows.
+ */
+void fl_application_set_default_window_title(FlApplication* application,
+                                             const gchar* window_title);
+
+/**
+ * fl_application_set_default_window_size:
+ * @application: an #FlApplication.
+ * @width: width in pixels or -1 to set the default width.
+ * @height: height in pixels or -1 to set the default height.
```

Comment:
What handles the `-1` logic? I don't see this in `fl_application_set_default_window_size`'s implementation

Reply:
Yes, it's just the standard GTK behavior for setting window sizes. So I figured we should document it here. Again, like the `fl_application_set_default_window_title` I think this will be deprecated in the multi-window case.

Reply:
In trying to write the documentation I decided that I should just implement the planned support for overriding window creation - then the application can set any properties of the window they require including title and size.